sharkfuser/samples/convolution/conv_wgrad_nhwc_krsc.cpp

```diff
@@ -0,0 +1,125 @@
+// Copyright 2025 Advanced Micro Devices, Inc.
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <fusilli.h>
+
+#include "utils.h"
+
+#include <catch2/catch_test_macros.hpp>
+#include <cstdint>
+#include <memory>
+#include <tuple>
+#include <unordered_map>
+#include <vector>
+
+using namespace fusilli;
+
+TEST_CASE("Convolution wgrad; DY/X (NHWC), DW (KRSC); 1x1; no padding",
+          "[conv][graph]") {
+  int64_t n = 4, c = 8, h = 8, w = 8, k = 16, r = 1, s = 1;
+
+  auto build_new_graph = [=](const Handle &handle) {
+    auto graph = std::make_shared<Graph>();
+    graph->setName("conv_wgrad_sample_nhwc_krsc_1x1_nopad");
+    graph->setIODataType(DataType::Float).setComputeDataType(DataType::Float);
+
+    auto DY = graph->tensor(TensorAttr()
+                                .setName("dy")
+                                .setDim({n, k, h, w})
+                                .setStride({k * h * w, 1, k * w, k})); // NHWC
+
+    auto X = graph->tensor(TensorAttr()
+                               .setName("x")
+                               .setDim({n, c, h, w})
+                               .setStride({c * h * w, 1, c * w, c})); // NHWC
+
+    auto wgradAttr = ConvWGradAttr()
+                         .setStride({1, 1})
+                         .setPadding({0, 0})
+                         .setDilation({1, 1})
+                         .setName("conv_wgrad");
+
+    auto DW = graph->convWGrad(DY, X, wgradAttr);
+    DW->setName("dw")
+        .setDataType(DataType::Float)
+        .setOutput(true)
+        .setDim({k, c, r, s});
+
+    // Validate, infer missing properties
+    REQUIRE(isOk(graph->validate()));
```

Comment:
we now use `FUSILLI_REQUIRE_OK` which is slightly better with providing the error messages when it fails.. 

https://github.com/nod-ai/shark-ai/pull/2556